### DISCUSSION TITLE
Trigger identify from peer_id

### DISCUSSION THREAD

**Discussion by vbhattaccmu** [libp2p/rust-libp2p#5537] [Q&A] [answered]
Is it possible to trigger an `identify::Event::Received` event in a p2p event loop without dialing a peer multiaddress. 
The closest API I came across is the `identify.push(vec![peer_id])`. But it doesn't seem to initiate identify protocol .

**Comment by dariusc93** [comment, marked as answer]
Hey! Assuming I am understanding your question correctly, at the time I dont believe that there is anything in spec or in the API that would allow one to request an identify from a peer. You could probably do this outside of the behaviour by sending a request to the peer using your own protocol (assuming both peers are using it) and then have that peer push to your peer id.
